src/components/ClaimENSPage/ClaimENSPage.tsx

```diff
@@ -22,7 +22,7 @@ export default class ClaimENSPage extends React.PureComponent<Props, State> {
 
   handleManaApprove = async () => {
     const { allowance, onAllowMana } = this.props
-    const manaToAllow = isAllowed(allowance) ? 0 : getMaximumValue()
+    const manaToAllow = isEnoughClaimMana(allowance) ? 0 : getMaximumValue()
```

Comment:
If the allowance is `> 0`, the action should be: 
1) Clean the allowance
2) Set a new allowance

I think that we should change this to:

```typscript
    const manaToAllow = allowance > 0 ? 0 : getMaximumValue()
```